WalletWasabi/Transactions/TransactionBuilding/SmartCoinSelector.cs

```diff
@@ -29,23 +30,54 @@ public IEnumerable<ICoin> Select(IEnumerable<ICoin> coins, IMoney target)
 				throw new InsufficientBalanceException(targetMoney, available);
 			}
 
-			// Group coins by scriptPubKey in order to treat them all as one.
-			var coinsByScriptPubKey = UnspentCoins
+			var uniqueClusters = UnspentCoins
+				.Select(coin => coin.Clusters)
+				.Distinct();
```

Comment:
This doesn't work. Cluster doesn't implement IEquitable

Reply:
It works. `Distinct` doesn't require `IEquitable`, otherwise the API would enforce it with a constrain `where T: IEquitable`. This is because reference types are comparable by reference by default and `IEquitable` has to be implemented _only_ when the default behavior has to be changed.  

```c#
var a = new { s="hello" };
var b = new { s="bye" };
var c = a;
new[]{ a, b, c }.Distinct().ToArray()
[{ s = "hello" }, { s = "bye" }]
```

Reply:
Yes, but also:

```
var a = new { s="hello" };
var b = new { s="hello" };
var c = a;
new[]{ a, b, c }.Distinct().ToArray()
[{ s = "hello" }, { s = "hello" }]
```

Reply:
Anyhow, I implemented it. At the very least it doesn't hurt.